sympy/physics/continuum_mechanics/beam.py

```diff
@@ -491,6 +501,55 @@ def apply_support(self, loc, type="fixed"):
         else:
             return reaction_load, reaction_moment
 
+    def apply_hinge(self, loc):
+        """
+        This method applies a hinge in a particular beam object.
+
+        Parameters
+        ----------
+        loc : Sympifyable
+            Location of point at which hinge is applied.
+
+        Examples
+        ========
+        There is a beam of length 15 meters. Pin supports are placed at distances
+        of 0 and 10 meters. There is a fixed support at the end. There are two hinges
+        in the structure, one at 5 meters and one at 10 meters. A pointload of magnitude
+        10 kN is applied on the hinge at 5 meters. A distributed load of 5 kN works on
+        the structure from 10 meters to the end.
+
+        Using the sign convention of upward forces and clockwise moment
+        being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(15, E, I)
+        >>> r0 = b.apply_support(0, type='pin')
+        >>> r10 = b.apply_support(10, type='pin')
+        >>> r15, m15 = b.apply_support(15, type='fixed')
+        >>> b.apply_hinge(5)
+        >>> b.apply_hinge(12)
+        >>> b.apply_load(-10, 5, -1)
+        >>> b.apply_load(-5, 10, 0, 15)
+        >>> b.solve_for_reaction_loads(r0, r10, r15, m15)
+        >>> b.reaction_loads
+        {M_15: -75/2, R_0: 0, R_10: 40, R_15: -5}
+        >>> b.bending_moment()
+        -9625*SingularityFunction(x, 5, -1)/24 + 10*SingularityFunction(x, 5, 1)
+        - 40*SingularityFunction(x, 10, 1) + 5*SingularityFunction(x, 10, 2)/2
+        + 1875*SingularityFunction(x, 12, -1)/16 + 75*SingularityFunction(x, 15, 0)/2
+        + 5*SingularityFunction(x, 15, 1) - 5*SingularityFunction(x, 15, 2)/2
+
+        """
+        loc = sympify(loc)
+        rotation_jump = Symbol('P_'+str(loc))
```

Comment:
Is this a symbol that the user may need to access? I don't see where it appears in the above example. In general, we should not create symbols internally, users should have to pass them in so that they have access to the variable we should make them accessible.

Reply:
In my opinion this is not a symbol that the user may need to access. This symbol describes the unknown rotation of the hinge in the beam. This is normally not something the user is interested in. It is necessary to create the symbol to describe the unknown rotation jump that is solved when the user calls `solve_for_reaction_loads`. The symbol is assigned a value in this method and that value is substituted into the load equation. 

Of course it is possible to handle this unknown in the same way the unknown reaction forces are handled in `apply_support`. The reason I have chosen not to do this in this PR is because this would make using the method a bit more difficult. This way the user would have to undertake more steps and the chance of entering the wrong `*reactions` into `solve_for_reaction_loads` would increase. 

I have taken both sides into consideration and thought this was the best way to handle this. As in my experience, you are almost never interessed in the value of rotation in a hinge.

Reply:
Now that I think about it there might be a way to have acces to the symbol but not have to pass it into  `solve_for_reaction_loads`. I will try to implement this and push when it works. This will probably be later today.